Sync storage with make up flow

KevinMind opened the pull request:
Relates to: mozilla/addons#15066
Child of: https://github.com/mozilla/addons-server/pull/22781

### Description

Sync the `storage` director with make initialize command. 1) when you dump/load a db backup it will store/load the storage directory as well (correctly) 2) when seeding the db it will sync the storage directory as well 3) when syncing the storage directory it will clear the directory first to ensure pristine backups 4) remove the storage volume and just mount the repo and map the files correctly to static or storage directories.

### Context

Currently `data_seed` is semi broken as it does not sync the storage directory when creating a snapshot.

### Testing

run a clean init

> NOTE: this will remove your existing DB so create a backup if you want to restore after

```bash
make up INIT_CLEAN=True
```

Expect the storage directory to be cleared and populated with the data from addons created during the seed.

run a backup and restore.

1. have a running environment
2. upload an addon
3. create a backup `make data_dump`
4. upload another addon
5. run a restore `make data_load`
6. expect the files associated with your 1st addon to be restored but not the 2nd.

### Checklist

- [X] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [X] Successfully verified the change locally.
- [X] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add before and after screenshots (Only for changes that impact the UI).
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

KevinMind commented on src/olympia/search/management/commands/reindex.py:
```diff
@@ -141,6 +146,10 @@ def handle(self, *args, **kwargs):
             )
         self.stdout.write('Starting the reindexation for %s.' % alias)
 
+        if kwargs['skip_if_exists'] and ES.indices.exists(index=alias):
```
TIL. I'm happy to move this to a separate PR. It is not really related to storage you're right. It's just a perf benefit for a large number of the times your run initialize

KevinMind replied:
Spun off https://github.com/mozilla/addons-server/pull/22819

diox commented:
Saw that `service "web" refers to undefined volume storage: invalid compose project` error in CI ?

diox commented:
Interesting issue I ran into:
- I was logged in
- I ran the dump
- I re-initialized my db
- I logged in again (creating a new user, since this is a fresh db)
- I ran the restore
- I refreshed the /developers/ page, and I get a redirect to the login flow, except I got that error:
```
SessionInterrupted at /api/v5/accounts/login/start/

The request's session was deleted before the request completed. The user may have logged out in a concurrent request, for example.
```

Digging into it, this appears to be because we're using cached db session engine and Django is using "new" cache with "old" data, so that doesn't quite work properly. Indeed, doing a `from django.core.cache import cache; cache.clear()` fixed the issue. 

Since this PR is about syncing up storage with the db, maybe it makes sense to address this here as well - That error can manifest in other ways, and it's a very similar underlying problem of cache not being consistent with the db backup we just loaded. I assume we should just clear the cache after a `data_load` to fix this.

diox commented:
> expect the files associated with your 1st addon to be restored but not the 2nd.

I can still see files from my second add-on in the storage directory (even though it doesn't exist in the db anymore)

KevinMind commented:
@diox:
- the first issue about invalid volume is a bug from rebasing, now fixed.
- updated the instructions you are right it is data_load
- added the clear cache method in data_load after we do the db restore. tried reproducing your issue and could not
- added logic to clear the storage dir before syncing from the backup that should fix the issue of stale data post data load
